### PR TITLE
TFFT rework

### LINKED ISSUE #6801: TFFT should be able to use other glass types
#### Which modpack version are you using?
2.0.9.1a dev
#
It would look fancier if we could use other glass types. We could also restrict the upgrades by glass tier if we wanted (though I probably couldn't do that part). Would anyone care if it was just any glass? Or just glass tier EV+?

### LINKED ISSUE #8053: Ender Conduit cannot extract fluids from T.F.F.T Multi I/O Hatch
#### Which modpack version are you using?
2.1.0.0 with KekzTech 0.5.5
#
#### If in multiplayer; On which server does this happen? 
MP/SP
#
#### What did you try to do, and what did you expect to happen?
I want to extract all fluids from TFFT to supertanks with ender conduit.
#
#### What happened instead? (Attach screenshots if needed)
Only the fluid in #0 slot can be extracted. But ME storage bus can access all the fluids. (Both mentioned in the TFFT quest)
#
#### What do you suggest instead/what changes do you propose?
Fix it
#
#### What is your GTNH Discord username?
r4phael#8493

### LINKED ISSUE #10061: T.F.F.T. Voiding stored fluids and various other bugs
### Your GTNH Discord Username

DacerFanAccount#4757

### Your Pack Version

2.1.2.3qf

### Your Server

private Server

### Type of Server

Vanilla Forge

### Your Expectation

Used a T.F.F.T. as a Fluid Storage
Imported all, previously in super tanks, stored fluids. Added a Multi I/O Hatch at the bottom to fill the various fluids in.
The Multi I/O Hatch does not output into ender fluid conduits.
Because of that i placed 25 output hatches (LV) adjacent to the center pillar, replacing the reinforced glass structure within that Multiblock. Then i locked every single available fluid to one fluid hatch (each) on the outside and then removed the circuit such that the Tank will refill all hatches on its own, because the T.F.F.T. doesnt output anything to hatches locked with a "null circuit" present (a circuit with a number of an ID thats not been filled in the T.F.F.T. yet), i changed the circuit to a number of a present fluid. this lead to the hatch filling up but also voiding all fluid contents of that respective ID.

### The Reality

Observation 1: The Multi I/O Hatch does not output into ender fluid conduits (perhaps a bug, but i dont know if that may be intended behaviour).

Observation 2: The T.F.F.T. does not output to locked hatches unless a fluid has been present inside that hatch after locking it
(replicate by filling a T.F.F.T. with for example water, then lock the T.F.F.T. from outputting anything by placing a circuit with a different ID from all other fluids, place a hatch, lock it to water, remove circuit, the T.F.F.T. will not refill the hatch until you place a water cell inside of the hatch first)

Observation 3: Placing a circuit with a fluid's ID (for example 22 for Iron-III-Chloride) inside the Controller and then taking a Cell out of the respective output hatch leads to the T.F.F.T. overwriting the previously stored fluid ID with NULL, essentially voiding the Fluid

### Your Proposal

-make tfft I/O hatch output to ender conduits pls
-make tfft less evil such that other people wont suffer the tragic fate of it voiding 2 million mB of fluorine while setting up
-make tfft output to locked hatches properly

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

### LINKED ISSUE #11654: TFFT
### Your GTNH Discord Username

FoKuCNiK

### Your Pack Version

last version

### Your Proposal

TFFT no hologram? I think this needs to be fixed, this is not the most useless multiblock and it should have a hologram like everyone else.

### Your Goal

HOLOGRAM

### Your Vision

HOLOGRAM

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this change please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members think it's inappropriate.
- [X] I believe this feature would make the pack better.

### PR DESCRIPTION
**Pull request by DianeXD**
Basically the same behavior as the previous TFFT, with a more flexible structure.
tfft
desc1
desc2

close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/11654
close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/10061
close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/8053
close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/6801

### CODE REVIEW

File: src/main/java/common/tileentities/GTMTE_TFFT.java [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+                                            ofBlockUnlocalizedName("Thaumcraft", "blockCosmeticOpaque", 2, false),
+                                            BorosilicateGlass.ofBoroGlassAnyTier()))
+                    .addElement(
+                            'g',
+                            ofChain(
+                                    ofBlockUnlocalizedName("IC2", "blockAlloyGlass", 0, true),
+                                    ofBlockUnlocalizedName("Thaumcraft", "blockCosmeticOpaque", 2, false),
+                                    BorosilicateGlass.ofBoroGlassAnyTier()))
+                    .addElement('f', ofChain(new IStructureElement<GTMTE_TFFT>() {
+                        @Override
```
Comment (minecraft7771) [inline src/main/java/common/tileentities/GTMTE_TFFT.java]: I think if you'd implement the structure element like here https://github.com/GTNewHorizons/GoodGenerator/blob/master/src/main/java/goodgenerator/blocks/tileEntity/NeutronActivator.java#L450 You'd getter cleaner code and also less code that is just duplicated from the base class of structurelib.

**Review by minecraft7771 — CHANGES_REQUESTED** [review]

File: src/main/java/common/tileentities/GTMTE_TFFT.java (hunk 1 above)
Comment (DianeXD) [reply src/main/java/common/tileentities/GTMTE_TFFT.java]: I'm just following the LSC style, but it's a good idea.
But I don't understand the duplicate, it's just overriding some methods

Comment (minecraft7771) [reply src/main/java/common/tileentities/GTMTE_TFFT.java]: What I mean is the code within the override. Its mostly duplicated from the overwritten method with your values injected. However functions like addTieredBlock already exist where you can specify all valid tiered blocks. If you want I could also implement this change into this PR if you have trouble following what I mean.

Comment (DianeXD) [reply src/main/java/common/tileentities/GTMTE_TFFT.java]: Ah, I see, but tfft storage blocks can be mixed, not sure if this method works

Comment (minecraft7771) [reply src/main/java/common/tileentities/GTMTE_TFFT.java]: Sure, you can input a list of possible blocks + meta if the blocks are not the same.

Comment (minecraft7771) [reply src/main/java/common/tileentities/GTMTE_TFFT.java]: Ah I see what you mean now. Ok then I can approve now.

**Review by minecraft7771 — APPROVED** [review]

**Pull request merged**